src/wrapper/train_spawner.py

```diff
@@ -0,0 +1,59 @@
+from typing import List
+
+from src.interlocking_component.route_controller import RouteController
+from src.wrapper.simulation_object_updating_component import (
+    SimulationObjectUpdatingComponent,
+)
+from src.wrapper.simulation_objects import Platform, Train
+
+
+class TrainSpawner:
+    """A factory to construct trains wich drive through the simulation"""
+
+    def __init__(
+        self,
+        updater: SimulationObjectUpdatingComponent,
+        route_controller: RouteController,
+    ):
+        self._updater: SimulationObjectUpdatingComponent = updater
+        self.route_controller: RouteController = route_controller
+
+    def spawn_train(
+        self, identifier: str, timetable: List[str], train_type: str
+    ) -> bool:
+        """Spawns a new train in the simulation
+
+        :param identifier: The id of the new train
+        :param timetable: The stations the train drives along (as string-ids)
+        :param train_type: the type of the train (corresponds to a sumo train type)
+        :return: if the spawning was successful
+        """
+        timetable = self._convert_timetable(timetable)
+
+        assert len(timetable) >= 2
```

Comment:
Couldn't there theoretically be a train that just stands there for the entire length of the simulation?

Reply:
Yes, but SUMO needs a route associated with a vehicle, so just one platform wouldn't be enough. If we want to model that, we would need to *SUMO train by using a route with `start==end` and `max_speed==0` or something equivalent (freezing the train)